app/frontend/schools/courses/components/curriculum_editor/CurriculumEditor__TargetDetailsEditor.res

```diff
@@ -583,30 +589,44 @@ let methodOfCompletionSelection = polyMethodOfCompletion =>
   | #TakeQuiz => TakeQuiz
   | #VisitLink => VisitLink
   | #MarkAsComplete => MarkAsComplete
+  | #SubmitForm => SubmitForm
   }
 
 let methodOfCompletionButton = (methodOfCompletion, state, send, index) => {
   let buttonString = switch methodOfCompletion {
   | #TakeQuiz => t("take_quiz")
   | #VisitLink => t("visit_link")
   | #MarkAsComplete => t("mark_as_complete")
+  | #SubmitForm => t("submit_form")
   }
 
   let selected = switch (state.methodOfCompletion, methodOfCompletion) {
   | (TakeQuiz, #TakeQuiz) => true
   | (VisitLink, #VisitLink) => true
   | (MarkAsComplete, #MarkAsComplete) => true
+  | (SubmitForm, #SubmitForm) => true
   | _anyOtherCombo => false
   }
 
   let icon = switch methodOfCompletion {
   | #TakeQuiz => quizIcon
   | #VisitLink => linkIcon
   | #MarkAsComplete => markIcon
+  | #SubmitForm => formIcon
   }
 
+  let disabled =
```

Comment:
I forgot to cleanup this as I initially thought of showing as first three options as greyed out when yes is selected but later sticked with current way. 

Will clean up this in upcoming commits

FYI @mahesh-krishnakumar